### PR TITLE
Replace of componentWillReceiveProps method on Agenda, Calendar, Calendar-List

### PR DESCRIPTION
**Pull request by AgustinJimenez**
According to react warning https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html

### CODE REVIEW

File: src/calendar-list/index.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ class CalendarList extends Component {
     this.listView.scrollToOffset({offset: scrollAmount, animated: false});
   }
 
-  componentWillReceiveProps(props) {
+  UNSAFE_componentWillReceiveProps(props) {
```
Comment (oliver-ni) [inline src/calendar-list/index.js]: probably not a good idea

Comment (AgustinJimenez) [reply src/calendar-list/index.js:142]: why?

File: src/agenda/index.js [hunk 2]
```diff
@@ -216,14 +216,14 @@ export default class AgendaView extends Component {
...
 
-  componentWillReceiveProps(props) {
+  static getDerivedStateFromProps(props, state) {
     if (props.items) {
-      this.setState({
-        firstResevationLoad: false
-      });
+      state.firstResevationLoad = false;
     } else {
       this.loadReservations(props);
```
Comment (Inbal-Tish) [inline src/agenda/index.js:223]: There is no access to 'this' in a static method

File: src/calendar-list/index.js (hunk 1 above)
Comment (Inbal-Tish) [reply src/calendar-list/index.js:142]: This method should be replaced as it's being deprecated soon

**Comment by Inbal-Tish** [comment]
@AgustinJimenez Thank you for the PR. Please check out PR #1356. Thanks!

**Pull request closed without merging**